### PR TITLE
Revise Unsqueeze op - tests

### PR DESCRIPTION
**Pull request by blesniewski**
### Details:
 - *Added visitor test*
 - *Added Serialization Single Layer Test*
 - *Separated empty axes (2nd input) cases in SLT only for Squeeze op (Unsqueeze requires non-empty axes)*
 - *Added op to trusted op list*

### Tickets:
 - *45908*
 - *53971*
 - *37490*

### CODE REVIEW

File: ngraph/test/backend/unsqueeze.in.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    auto axes_node =
+        make_shared<ngraph::op::Constant>(element::i64, Shape{2}, vector<int64_t>{1, 2});
+    auto squeeze = make_shared<op::v0::Unsqueeze>(data_node, axes_node);
+
+    auto function = make_shared<Function>(NodeVector{squeeze}, ParameterVector{data_node});
+    auto test_case = test::TestCase<TestEngine>(function);
+
+    auto data = vector<float>{1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f};
+    test_case.add_input(data);
+    test_case.add_expected_output<float>(Shape{4, 1, 1, 2}, data);
```
Comment (ggalieroc) [inline ngraph/test/backend/unsqueeze.in.cpp:29]: Are we sure the test cases are being run. I can't see call to run `test_case.run()`.

File: ngraph/test/visitors/op/unsqueeze.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+using namespace std;
+using namespace ngraph;
+using ngraph::test::NodeBuilder;
+
+TEST(attributes, unsqueeze_op)
+{
+    using namespace opset1;
+
+    auto param = make_shared<op::Parameter>(element::f32, Shape{4, 1, 4, 1, 8});
+    auto axex = make_shared<ngraph::op::Constant>(element::u64, Shape{2}, vector<int64_t>{1, 2});
```
Comment (ggalieroc) [inline ngraph/test/visitors/op/unsqueeze.cpp:22]: nitpick: `axes`

File: ngraph/test/backend/unsqueeze.in.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    auto axes_node =
+        make_shared<ngraph::op::Constant>(element::i64, Shape{2}, vector<int64_t>{1, -1});
+    auto squeeze = make_shared<op::v0::Unsqueeze>(data_node, axes_node);
+
+    auto function = make_shared<Function>(NodeVector{squeeze}, ParameterVector{data_node});
+    auto test_case = test::TestCase<TestEngine>(function);
+
+    auto data = vector<float>{1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f, 7.0f, 8.0f};
+    test_case.add_input(data);
+    test_case.add_expected_output<float>(Shape{4, 1, 2, 1}, data);
```
Comment (ggalieroc) [inline ngraph/test/backend/unsqueeze.in.cpp:44]: Same as in comment above.

File: ngraph/test/backend/unsqueeze.in.cpp (hunk 1 above)
Comment (blesniewski) [reply ngraph/test/backend/unsqueeze.in.cpp:29]: 🤦

**Review by ggalieroc — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**